src/popup/utils/utils.ts

```diff
@@ -480,16 +480,20 @@ export function openInNewWindow(url: string) {
 }
 
 export function calculateFontSize(amountValue: BigNumber | number) {
-  const amount = new BigNumber(amountValue);
-  if (amount.isLessThanOrEqualTo(999999)) {
+  const amountLength = amountRounded(amountValue).length;
+
+  if (amountLength <= 6) {
     return '18px';
   }
-  if (amount.isLessThanOrEqualTo(999999999)) {
+
+  if (amountLength <= 9) {
     return '16px';
   }
-  if (amount.isLessThanOrEqualTo(999999999999)) {
+
+  if (amountLength <= 12) {
     return '14px';
   }
+
   return '12px';
```

Comment:
Maybe instead of returns this approach could be slimmer:
```
let size = '12px';
if ( ... ) {
  size = ...
} else if ( ... ) {
  size = ...
} else if ( ... ) {
 ...
}
return size;
```